src/scripts/main.js

```diff
@@ -1,3 +1,32 @@
 'use strict';
 
-// write code here
+const listItems = [...document.querySelectorAll('li')];
+const employeesList = document.querySelector('ul');
+
+function convertToNumber(string) {
+  const salaryNumber = string.replace(/\$/g, '').replace(/,/g, '');
+
+  return +salaryNumber;
+}
+
+function sortBySalary(items) {
+  items.sort(
+    (a, b) =>
+      convertToNumber(b.dataset.salary) - convertToNumber(a.dataset.salary),
+  );
+
+  return items;
+}
```

Comment:
Bug: The sort function directly modifies the original array. This can lead to unexpected side effects if the original array is used elsewhere in the code. It's generally a better practice to create a new array and sort that, leaving the original array unmodified. This can be done by using the spread operator to create a new array before sorting it.